app/src/main/java/com/example/poste/activities/DashboardActivity.java

```diff
@@ -91,47 +78,99 @@ protected void onCreate(Bundle savedInstanceState) {
+                for (Folder folder : userFolders) {  // repopulate the map
+                    folderNameToIdMap.put(
+                            folder.getTitle(),
+                            folder.getId()
+                    );
+                }
+                // Fill folder view (Recycler View)
+                // Note: This is not the best way to do this, because it means every time
+                // DashboardActivity restarts, a new Adapter is created, but it works for now. The
+                // better way would be to use DiffUtil to calculate the difference between the old
+                // and new lists, and then update the RecyclerView accordingly using
+                // AdapterListUpdateCallback. This method is an attempt to keep the code simple
+                // For more information, see:
+                // https://developer.android.com/reference/androidx/recyclerview/widget/DiffUtil
+                // and
+                // https://developer.android.com/reference/androidx/recyclerview/widget/ListUpdateCallback
+                folderAdapter = new FolderAdapter(
+                        new FolderAdapter.ClickListener() {
+                            @Override
+                            public void onItemClick(int position, Folder model) {
+                                // Send to that folder's view
+                                PosteApplication.setSelectedFolder(model);
+                                Intent intent = new Intent(DashboardActivity.this, FolderViewActivity.class);
+                                startActivity(intent);
+                                // Don't finish this activity, so that we can come back to the
+                                // dashboard
+                            }
+
+                            @Override
+                            public void onItemLongClick(int position, Folder model) {
+                                // Prompt to share folder
+                                Toast.makeText(DashboardActivity.this, getString(R.string.share_folder), Toast.LENGTH_LONG).show();
+                            }
+                        },
+                        new ArrayList<>(userFolders)
+                );
+                folderRecyclerView.setAdapter(folderAdapter);
+                registerForContextMenu(folderRecyclerView);
+            }
+
+            /**
+             * This defines what will happen if the update fails; what we want the application
+             * to do in a failure case (e.g. if the API call fails).
+             * @param errorMessage The error message returned by the API call (or whatever we send
+             *                     the callback to)
+             */
+            @Override
+            public void onError(String errorMessage) {
+                Toast.makeText(
+                        DashboardActivity.this,
+                        "Unable to retrieve folders and posts, please try again.",
```

Comment:
Even though it is nice to get such a feedback as a user, right now the only way to correct the issue is to signout and login again (if im not wrong). Maybe we should consider implementing the refresh option we talked about. But obviously that is out of the scope of what you were working on.

Reply:
Could hit back, or a drag-down, yeah. Good input; might be valuable to create a User Story out of that though so it doesn't get lost in this PR